### PR TITLE
Add unsubscribe and preference link context to test emails

### PR DESCRIPTION
**Pull request by VenkataNeeharikaKondipati**
- Add unsubscribe and preference links to email profile
- This will ensure test emails have the appropriate links

_A summary of your pull request, including the what change you're making and why._

## Testing

- Tested sendgrid DA locally by firing a test event
- Tested in stage using traditional deploy: https://github.com/segmentio/eks-configuration/pull/5182

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
Hi @VenkataNeeharikaKondipati , 
thanks for raising this PR. Could you please add some details about what the PR is for please, and also provide info on how you tested the change?

I'm just back from Recharge PTO so have a lot to catch up on. It's unlikely this change will get deployed this week. Let's aim for next Tuesday 8-Aug. 

Thanks,
Joe

**Comment by VenkataNeeharikaKondipati** [comment]
> Hi @VenkataNeeharikaKondipati , thanks for raising this PR. Could you please add some details about what the PR is for please, and also provide info on how you tested the change?
> 
> I'm just back from Recharge PTO so have a lot to catch up on. It's unlikely this change will get deployed this week. Let's aim for next Tuesday 8-Aug.
> 
> Thanks, Joe

Hi Joe, this change is to add unsubscribe and preference links to test emails functionality from the Engage channels. I tested this change locally by sending a test event. We want to see if you can deploy this change this week as it is a fast follow for our GA release for 'subscription-groups' feature

File: packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts [hunk 1]
```diff
@@ -58,18 +58,19 @@ export class SendEmailPerformer extends MessageSendPerformer<Settings, Payload>
 
   getRecepients(): ExtId<Payload>[] {
     //if toEmail specified => send test email requested
-    if (this.payload.toEmail)
+    if (this.payload.toEmail) {
+      const externIdContext = this.payload?.externalIds && this.payload?.externalIds[0]
```
Comment (sumitbhanwala) [inline packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:62]: nit : externIdContext is missing an 'a'.

**Review by vikramkumar19 — COMMENTED** [review]
LGTM! However, you may want to correct the version tags.

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @VenkataNeeharikaKondipati this PR has been deployed.

**Pull request merged**